packages/readmodel/src/purpose-template/aggregators.ts

```diff
@@ -0,0 +1,413 @@
+      >(annotationSQL.answerId);
+      acc.set(answerId, annotationSQL);
+
+      return acc;
+    }, new Map<RiskAnalysisSingleAnswerId | RiskAnalysisMultiAnswerId, PurposeTemplateRiskAnalysisAnswerAnnotationSQL>());
+
+  const riskAnalysisTemplateAnswersAnnotationsDocumentsByAnnotationId =
+    riskAnalysisTemplateAnswersAnnotationsDocumentsSQL.reduce(
+      (acc, documentSQL) => {
+        const annotationId =
+          unsafeBrandId<RiskAnalysisTemplateAnswerAnnotationId>(
+            documentSQL.annotationId
+          );
+        const documentsSQL = acc.get(annotationId) || [];
+        // eslint-disable-next-line functional/immutable-data
+        documentsSQL.push(documentSQL);
+        acc.set(annotationId, documentsSQL);
+
+        return acc;
+      },
+      new Map<
+        RiskAnalysisTemplateAnswerAnnotationId,
+        PurposeTemplateRiskAnalysisAnswerAnnotationDocumentSQL[]
+      >()
+    );
+
+  const {
+    riskAnalysisTemplateSingleAnswers,
+    riskAnalysisTemplateMultiAnswers,
+  } = riskAnalysisTemplateAnswersSQL.reduce(
+    (acc, answerSQL) => {
+      const answerId = unsafeBrandId<
+        RiskAnalysisSingleAnswerId | RiskAnalysisMultiAnswerId
+      >(answerSQL.id);
+      const annotationSQL =
+        riskAnalysisTemplateAnswersAnnotationsByAnswerId.get(answerId);
+
+      const annotationDocumentSQL = annotationSQL
+        ? riskAnalysisTemplateAnswersAnnotationsDocumentsByAnnotationId.get(
+            unsafeBrandId(annotationSQL.id)
+          )
+        : undefined;
+
+      const answerAnnotation: RiskAnalysisTemplateAnswerAnnotation | undefined =
+        annotationSQL
+          ? {
+              id: unsafeBrandId(annotationSQL.id),
+              text: annotationSQL.text,
+              docs: annotationDocumentSQL
+                ? annotationDocumentSQL.map((doc) => ({
+                    id: unsafeBrandId(doc.id),
+                    name: doc.name,
+                    prettyName: doc.prettyName,
+                    contentType: doc.contentType,
+                    path: doc.path,
+                    createdAt: stringToDate(doc.createdAt),
+                  }))
+                : [],
+            }
+          : undefined;
+
+      return match(RiskAnalysisAnswerKind.parse(answerSQL.kind))
+        .with(riskAnalysisAnswerKind.single, () => ({
+          riskAnalysisTemplateSingleAnswers: [
+            ...acc.riskAnalysisTemplateSingleAnswers,
+            {
+              id: unsafeBrandId(answerSQL.id),
+              key: answerSQL.key,
+              ...(answerSQL.value[0] ? { value: answerSQL.value[0] } : {}),
+              editable: answerSQL.editable,
+              ...(answerAnnotation ? { annotation: answerAnnotation } : {}),
+              suggestedValues: answerSQL.suggestedValues || [],
+            } satisfies RiskAnalysisTemplateSingleAnswer,
+          ],
+          riskAnalysisTemplateMultiAnswers: [
+            ...acc.riskAnalysisTemplateMultiAnswers,
+          ],
```

Comment:
Can we remove the unnecessary spread here and in the other branch? It's creating a new array when it's not needed, which is a bit of wasted performance